cmd2/argparse_completer.py

```diff
@@ -577,16 +577,15 @@ def _complete_for_arg(self, arg_state: _ArgumentState,
         args = []
         kwargs = {}
         if isinstance(arg_choices, ChoicesCallable):
-            if arg_choices.is_method:
-                # The completer may or may not be defined in the same class as the command. Since completer
-                # functions are registered with the command argparser before anything is instantiated, we
-                # need to find an instance at runtime that matches the types during declaration
-                cmd_set = self._cmd2_app._resolve_func_self(arg_choices.to_call, cmd_set)
-                if cmd_set is None:
-                    # No cases matched, raise an error
-                    raise CompletionError('Could not find CommandSet instance matching defining type for completer')
+            # The completer may or may not be defined in the same class as the command. Since completer
+            # functions are registered with the command argparser before anything is instantiated, we
+            # need to find an instance at runtime that matches the types during declaration
+            self_arg = self._cmd2_app._resolve_func_self(arg_choices.to_call, cmd_set)
+            if self_arg is None:
```

Comment:
Do we still support standalone functions? If not, I think we should. If self is `None` here, we can pass the cmd2 app by default. This would require standalone functions to take `cmd2.Cmd` as the first parameter.

We also wouldn't need to move the utility completers if we do this.

Edit:
Nevermind, I look at the resolve method again. This should never be None.